Add SSH certificates documentation (#4592)
theforeman/foreman-documentation: guides/common/modules/con_using-ssh-certificates.adoc

:_mod-docs-content-type: CONCEPT

[id="using-ssh-certificates_{context}"]

> = Using SSH certificates

Reviewer: ```suggestion
= Configuring remote execution to use SSH certificates to authenticate a user connection
```

You can use SSH certificates if you want to avoid the Trust on First Use (TOFU) model.

> SSH certificates are signed by a Certificate Authority (CA), which allows centralized trust management and revocation capabilities.

Reviewer: ```suggestion
By default, {Project} follows the Trust on First Use (TOFU) model during a remote execution connection.
You can reconfigure your {SmartProxy} to use SSH certificates instead.
SSH certificates are signed by a Certificate Authority (CA), which allows centralized trust management and revocation capabilities.
```

You can configure SSH certificate authentication in two ways:

{SmartProxy} authentication::
The {SmartProxy} presents a signed certificate when connecting to managed hosts.
Hosts trust the CA that signed the certificate, eliminating the need to distribute individual SSH public keys to each host.

Host authentication::
Hosts present signed certificates when {SmartProxy} connects to them.

> This provides verification that {SmartProxy} is connecting to legitimate hosts, preventing man-in-the-middle attacks.

Reviewer: ```suggestion
This provides verification that {SmartProxy} is connecting to legitimate hosts, helping prevent man-in-the-middle attacks.
```

Author: Wouldn't it be "helping to prevent"?

> You can configure one or both of these authentication methods depending on your security requirements.

Reviewer: ```suggestion
You can configure SSH certificate authentication in the following ways.
You can configure one or both of these authentication methods depending on your security requirements:

{SmartProxy} authentication::
The {SmartProxy} presents a signed certificate when connecting to managed hosts.
Hosts trust the CA that signed the certificate, eliminating the need to distribute individual SSH public keys to each host.

Host authentication::
Hosts present signed certificates when {SmartProxy} connects to them.
This provides verification that {SmartProxy} is connecting to legitimate hosts, preventing man-in-the-middle attacks.
```

Avoiding "two" helps prevent the (relatively common) maintenance issues when a new option is added. And merging the intro and outro sentences IMO makes faster reading.
